torch/distributed/fsdp/fully_sharded_data_parallel.py

```diff
@@ -1455,6 +1508,20 @@ def _unshard(
         Postcondition: Each handle's ``FlatParameter`` 's data is the padded
         unsharded flattened parameter on the compute device.
         """
+        if self.limit_all_gathers:
+            queue = self._free_event_queue
+            events: List[FreeEvent] = []
+            if queue.queue_size > self._max_inflight_all_gather_size:
```

Comment:
wondering whether we should use queue_size or current runtime cached/allocated memory utilization %..? if we use queue_size, the _max_inflight_all_gather_size probably needs to be larger than the max of the fsdp instances' full parameter size?

Reply:
> if we use queue_size, the _max_inflight_all_gather_size probably needs to be larger than the max of the fsdp instances' full parameter size?

Yes, unless you want to have only 1 inflight all-gather at a time, which I suppose should never be desired. I think until we have a better understanding of the correlation between the cached/allocated/active memory stats and the fast CPU thread, it may be hard to develop a heuristic using those.

In practice, most `FlatParameter`s should be around the same size, so approximately, a count-based limiter and a size-based limiter should be about the same. I am still working on experiments to explore.